src/main/kotlin/no/nav/amt/deltaker/bff/testdata/TestdataService.kt

```diff
@@ -0,0 +1,123 @@
+import no.nav.amt.deltaker.bff.deltakerliste.Deltakerliste
+import no.nav.amt.deltaker.bff.deltakerliste.DeltakerlisteService
+import no.nav.amt.deltaker.bff.deltakerliste.tiltakstype.toInnhold
+import no.nav.amt.lib.models.arrangor.melding.EndringFraArrangor
+import no.nav.amt.lib.models.deltaker.Deltakelsesinnhold
+import no.nav.amt.lib.models.deltakerliste.tiltakstype.Tiltakstype
+import java.time.LocalDate
+import java.time.LocalDateTime
+import java.util.UUID
+
+const val TESTVEILEDER = "Z990098"
+const val TESTENHET = "0314"
+
+// Plutselig Lagsport
+const val TESTARRANGORANSATT = "fff9a665-cbde-4dbc-9ef9-deb8681a0d6f"
+
+class TestdataService(
+    private val pameldingService: PameldingService,
+    private val deltakerlisteService: DeltakerlisteService,
+    private val arrangorMeldingProducer: ArrangorMeldingProducer,
+    private val deltakerService: DeltakerService,
+) {
+    suspend fun opprettDeltakelse(opprettTestDeltakelseRequest: OpprettTestDeltakelseRequest): Deltaker {
+        val deltakerliste = deltakerlisteService.get(opprettTestDeltakelseRequest.deltakerlisteId).getOrThrow()
+        val forventetSluttdato = opprettTestDeltakelseRequest.startdato.plusMonths(3)
+        valider(
+            startdato = opprettTestDeltakelseRequest.startdato,
+            sluttdato = forventetSluttdato,
+            deltakerliste = deltakerliste,
+        )
+        val kladd = pameldingService.opprettKladd(
+            deltakerlisteId = opprettTestDeltakelseRequest.deltakerlisteId,
+            personident = opprettTestDeltakelseRequest.personident,
+        )
+        val deltakerId = kladd.id
+
+        delay(10)
+
+        val utkast = getUtkast(deltakerId, deltakerliste, opprettTestDeltakelseRequest)
+        pameldingService.upsertUtkast(utkast)
+
+        delay(100)
+
+        val endringFraArrangor = getEndringFraArrangor(
+            deltakerId = deltakerId,
+            startdato = opprettTestDeltakelseRequest.startdato,
+            sluttdato = forventetSluttdato,
+        )
+        arrangorMeldingProducer.produce(endringFraArrangor)
+
+        delay(100)
+
+        return deltakerService.get(deltakerId).getOrThrow()
+    }
+
+    private fun valider(
+        startdato: LocalDate,
+        sluttdato: LocalDate,
+        deltakerliste: Deltakerliste,
+    ) {
+        if (deltakerliste.tiltak.tiltakskode != Tiltakstype.Tiltakskode.ARBEIDSFORBEREDENDE_TRENING) {
+            throw IllegalArgumentException("Det er kun AFT som er støttet for testdeltakelser inntil videre")
+        }
+        if (startdato.isBefore(deltakerliste.startDato)) {
+            throw IllegalArgumentException("Kan ikke sette startdato tidligere enn gjennomføringens startdato")
+        }
+        if (deltakerliste.sluttDato != null && sluttdato.isAfter(deltakerliste.sluttDato)) {
+            throw IllegalArgumentException("Kan ikke sette sluttdato senere enn gjennomføringens sluttdato")
+        }
+    }
+
+    private fun getEndringFraArrangor(
```

Comment:
nit: mulig det bare er meg men 'get' er for meg et ord som er reservert til "noe som hentes" fra api eller fra database så måtte tenke meg om en ekstra gang. Spesielt når man bruker et engelsk nøkkelord i en ellers norsk kontekst og det norske ordet ikke er noe lengre enn det engelske. Den kunne for eksempel hete 'lagEndringFraArrangor'